src/mps/cumps.jl

```diff
@@ -58,15 +78,57 @@ function productCuMPS(::Type{T}, ivals::Vector{<:IndexVal}) where {T<:Number}
   end
   return MPS(As,0,2)
 end
-productCuMPS(ivals::Vector{<:IndexVal}) = productCuMPS(Float64, ivals)
+cuMPS(ivals::Vector{<:IndexVal}) = cuMPS(Float64, ivals)
 
-function productCuMPS(::Type{T}, sites,
-                      states) where {T<:Number}
-  if length(sites) != length(states)
-    throw(DimensionMismatch("Number of sites and and initial states don't match"))
+function cuMPS(::Type{T}, sites::Vector{<:Index}, states_) where {T<:Number}
```

Comment:
Why did this function need to become longer?

It seems like there is a lot of code repitition with ITensors.jl, is there a way we could factorize out parts of CPU code in ITensors.jl so we can just reuse it here?

Reply:
Another thing I was wondering, could we just define this with something like `cuMPS(args...; kwargs...) = cu(MPS(args...; kwargs...))`? I know for efficiency reasons it may be some work to transfer to GPU, but these operations are probably not efficient on GPU anyway.

Reply:
This became longer because it's the fallback for all the following `cuMPS` methods, IIRC. I agree that we should think about refactoring this or perhaps making the CPU code more general.